### ISSUE TITLE
Tip: easy auto-update of Vim-Plug itself

### ISSUE THREAD

**Issue by davidstosik** [junegunn/vim-plug#1059]
Hello there, this is not your usual issue reporting a problem or requesting a feature, but as I didn't find any better place where to but this suggestion (likely the wiki), I though I'd post it here.

The story starts when I decided to reinstall my setup, including my dotfiles, and realized that I had not updated the `plug.vim` file in years. Sure, the process is simple, and I could have simply downloaded the new file and replaced it, but it's manual, and I know I won't remember.
I strive to have an up-to-date setup, with minimal manual steps involved, and the suggested way to install `vim-plug` became a bit of a frustration. Digging a bit, I figured out how `vim-plug` can update itself, as long as you remember to update your Vim plugins with `:PlugUpdate`.

Here's how I installed `vim-plug`:

```console
$ mkdir -p "$HOME/.vim/autoload" "$HOME/.vim/plugged"
$ git clone https://github.com/junegunn/vim-plug.git "$HOME/.vim/plugged/vim-plug"
$ ln -s "$HOME/.vim/plugged/vim-plug/plug.vim" "$HOME/.vim/autoload"
$ cat <<VIMRC >> $HOME/.vimrc
heredoc> call plug#begin('~/.vim/plugged')
heredoc> Plug 'junegunn/vim-plug'
heredoc> call plug#end()
heredoc> VIMRC

```

In short:
- Clone the `vim-plug` GitHub repository in the folder where `vim-plug` will install plugins.
- Create a symbolic link in Vim's autoload directory, pointing to the above repository's `plug.vim` file.
- Add `junegunn/vim-plug` to the list of plugins managed by `vim-plug`.
- Now, Vim autoloads `plug.vim` from a git repository, and `vim-plug` updates itself every time one runs `:PlugUpdate`.

I'm probably not the first to think of this, but didn't find any documentation in that sense (the closest I found was how it is possible to add `Plug 'junegunn/vim-plug'` to one's `.vimrc` file in order to get access to `vim-plug` Vim help), so I hope it will be helpful.

**Comment by janlazo** [comment]
https://github.com/junegunn/vim-plug/wiki/faq#shouldnt-vim-plug-update-itself-on-plugupdate-like-vundle

**Comment by davidstosik** [comment]
Thanks! Guess I haven't been looking hard enough... 😅
